src/palace.pyx

```diff
@@ -119,6 +120,18 @@ def query_extension(name: str) -> bool:
     return devmgr.query_extension(name)
 
 
+def current_context() -> Optional[Context]:
+    """Return the context that is currently used."""
+    if alure.Context.get_current() == <alure.Context> nullptr:
+        return None
+    cdef Context current = Context.__new__(Context)
+    current.impl = alure.Context.get_current()
```

Comment:
There's no point of calling `alure.Context.get_current()` twice.  As defined in `alure2.h`, `bool` of  `alure::Context` is the nonnullity, thus all you have to do is just `if not current: return None`.  Also in the line below use `__new__` to create the empty device, just for consistency (ugh why isn't @9a24f0 done with #38?).  Remember to `fixup` the changes (`git rebase -i HEAD~2`).

After this, use the **Rebase and merge** button on Github to merge the changes.  Then on branch master pump the version (in `setup.cfg`, to 0.0.10 I think, you can do it as part of the `fixup`) [draft a new pre-release](https://github.com/McSinyx/palace/releases) with `sdist` (`./setup.py sdist`) attached, wait for the wheels to get pushed to PyPI by Travis, then install it using pip.  Switch to `gh-pages` branch, update the version (in `src/conf.py`), generate the new documentation and push.  That's all the work for a new release lmao.

And don't forget to document how the context returned by `current_context` have the wrong message handler in a new issue.  Pseudo test:
```python
with context:
    assert current_context().message_handler is context.message_handler
```

Reply:
Test passed, feel free to **Rebase and merge** when you're ready.